Upped rebar warnings, tooling and deps. Fixed errors.

blt opened the pull request:
This commit doeesn't change the operation of habenero. It does update it's
dependencies and turn on some compile-time error checking. Well, compile
time-ish.

Signed-off-by: Brian L. Troutwine brian.troutwine@adroll.com

Commit:
Upped rebar warnings, tooling and deps. Fixed errors.

This commit doeesn't change the operation of habenero. It does update it's
dependencies and turn on some compile-time error checking. Well, compile
time-ish.

Signed-off-by: Brian L. Troutwine <brian.troutwine@adroll.com>

james-kelly commented on src/habanero_http_worker.erl:
```diff
@@ -161,25 +161,8 @@ call(undefined, _State) ->
-                  end,
-    Result.
-
+call({erlang, {Module, Function}}, #state{context = C} = _State) ->
```
This effectively removes support for JS in handlers. The support /was/ tentative, I'm ambivalent about pulling it out, but was this intentional?

blt replied:
Ah, good call. 

All the added xref and undefined function checks fail without `erlang_js` being enabled by default in the rebar.config. They rightly declare that the functions are undefined. I pulled this out while doing the work, intending it to be temporary, but forgot that I'd done it. 

For all the added compiler warnings to be in place this JS code needs to:
- have `erlang_js` enabled by default
- be commented out 
- be removed

I didn't investigate how hard it would be to enable `erlang_js` by default; I could do that. Thoughts?

Commit:
Reintroduce erlang_js.

I had some issues between R16 and R15 with starting the deps of habanero. This
commit introduces a recursive form of the common ensure_deps and relies on a
static habanero.app. That is not _strictly_ necessary but it does make debugging
this sort of startup issue more simple.

Signed-off-by: Brian L. Troutwine <brian.troutwine@adroll.com>
